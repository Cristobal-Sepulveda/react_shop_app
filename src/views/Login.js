import React, { useState } from "react";
import { Alert, View, Text, Button, TextInput, StyleSheet } from "react-native";
import AsyncStorage from '@react-native-async-storage/async-storage';




const Login = ({navigation}) =>{
    const [user, setUser] = useState("");
    const [passwd, setPasswd] = useState("");
    
    const navigateToNextView = async() => {
        if(user == "" || passwd == ""){
            Alert.alert("Alerta", "Debe Completar ambos campos antes de intentar iniciar sesión",)
        }else{
            try {
                await AsyncStorage.setItem('user', user);
                navigation.replace('Home')
            } catch (error) {
                    
            }
        }
    }

    const cleanData = () => {
        setUser("")
        setPasswd("")
    };

    return (
        <View style = {styles.container}>
            <View style = {styles.loginCard}>
                <Text>Bienvenido</Text>
                <TextInput style = {styles.textInput} placeholder="Ingrese Usuario" value={user} onChangeText={setUser} />
                <TextInput style = {styles.textInput} secureTextEntry = {true} placeholder="Ingrese Clave" value={passwd} onChangeText={setPasswd}/>
                <View style = {styles.viewButtons} >
                    <Button title = "Iniciar sesión" onPress={() => navigateToNextView(navigation, user,passwd)}/>
                    <Button title = "Limpiar Campos" onPress={()=> cleanData()}/>                                          
                </View>
            </View>
        </View>
    )
};


export default Login;

const styles = StyleSheet.create({
    container:{
        backgroundColor: "#dddddd",
        flex: 1,
        justifyContent: "center",
        alignItems: "center",
    },
    textInput:{
        color: "#000",
        width: 250,
        height: 40,
        marginTop: 16,
        borderWidth: 1,
        borderRadius: 10,
        padding: 8
    },
    viewButtons:{
        flexDirection: "row",
        marginTop: 16
    }
})
