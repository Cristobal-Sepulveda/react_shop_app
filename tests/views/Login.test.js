import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import Login from '../../src/views/Login';
import setupTests from '../enzymeSetup/setupTests'

 describe('Probamos el componente <Login />', () => {
  let wrapper = shallow(<Login />);

  beforeEach(() => {
         wrapper = shallow(<Login />);
  });

  test('debería mostrar <Login /> correctamente ', () => {
        expect(wrapper).toMatchSnapshot();
  });

  // test('debe de mostrar el valor por defecto de 100', () => {
  //        const wrapper = shallow(<CounterApp value={100} />);
  //        const counterText = wrapper.find('h2').text().trim();
  //        expect(counterText).toBe('100');
  // });

  // test('debe incrementar con el botón +1', () => {
  //        wrapper.find('button').at(0).simulate('click');
  //        const counterText = wrapper.find('h2').text().trim();
  //        expect(counterText).toBe('11');
  // });

  // test('debe decrementar con el botón -1', () => {
  //  wrapper.find('button').at(2).simulate('click');
  //        const counterText = wrapper.find('h2').text().trim();
  //        expect(counterText).toBe('9');
  // });

  // test('debe de colocar (...) el valor con el botón reset', () => {
  //   const wrapper = shallow(<CounterApp value={105} />);
  //   wrapper.find('button').at(0).simulate('click');
  //   wrapper.find('button').at(1).simulate('click');
  //   const counterText = wrapper.find('h2').text().trim();
  //   expect(counterText).toBe('105');
  // });
});
