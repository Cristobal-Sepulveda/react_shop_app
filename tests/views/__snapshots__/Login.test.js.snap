// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probamos el componente <Login /> debería mostrar <Login /> correctamente  1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#dddddd",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View>
    <Text>
      Bienvenido
    </Text>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Ingrese Usuario"
      rejectResponderTermination={true}
      style={
        Object {
          "borderRadius": 10,
          "borderWidth": 1,
          "color": "#000",
          "height": 40,
          "marginTop": 16,
          "padding": 8,
          "width": 250,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Ingrese Clave"
      rejectResponderTermination={true}
      secureTextEntry={true}
      style={
        Object {
          "borderRadius": 10,
          "borderWidth": 1,
          "color": "#000",
          "height": 40,
          "marginTop": 16,
          "padding": 8,
          "width": 250,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      style={
        Object {
          "flexDirection": "row",
          "marginTop": 16,
        }
      }
    >
      <Button
        onPress={[Function]}
        title="Iniciar sesión"
      />
      <Button
        onPress={[Function]}
        title="Limpiar Campos"
      />
    </View>
  </View>
</View>
`;
